pepsico/comparison.py

```diff
@@ -1,14 +1,147 @@
 import xarray as xr
+import os
+import re
+
+
+ds_zarr = '/data/remic/mydatafiles/zarr/test/hurs'
+ds_nc_dir = '/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global/daily/historical/MPI-ESM1-2-HR/'
+
+#only using NetCDF files that contain 'hurs' variable
+def contains_hurs(filename):
+    return 'hurs' in filename
+
+#extracting years from nc file
+def extract_years(filename):
+    match = re.search(r'(\d{4})_(\d{4})\.nc$', filename)
+    if match:
+        start_year = int(match.group(1))
+        end_year = int(match.group(2))
+        return start_year, end_year
+    else:
+        raise ValueError("Unable to find years")
+
+def find_nearest_indices(dataset, X, Y):
```

Comment:
sel has a nearest method option

https://docs.xarray.dev/en/stable/generated/xarray.DataArray.sel.html

that I think you could use instead

Reply:
Thank you for the great suggestion, I have pushed the modified code.